### PR TITLE
Hbrefactor

### PR DESCRIPTION
**Pull request by JPLaitinen**
Checked this with test app (only rewarded though will see how I test Interstatial)
There are quite lot of code refactored here. Interstatial and Rewarded are so similar (you will see once you review it) that I created base class to remove code duplication. I don't know if I should add this to our master for now it's on our header bidding branch

### CODE REVIEW

**Comment by JPLaitinen** [comment]
Ok now tested and it works with Interstitial ad too

File: UnityAds/src/main/java/com/mopub/mobileads/UnityRouter.java [hunk 1]
```diff
@@ -31,7 +33,12 @@ static void initUnityAds(Map<String, String> serverExtras, Context context, IUni
         }
         initMediationMetadata(context);
 
-        boolean testMode = false;
+        boolean testMode = true;
```
Comment (MileyHollenberg) [inline UnityAds/src/main/java/com/mopub/mobileads/UnityRouter.java:36]: Default should be false

File: UnityAds/src/main/java/com/mopub/mobileads/UnityRouter.java [hunk 2]
```diff
@@ -67,6 +74,11 @@ static void initMediationMetadata(Context context) {
         mediationMetaData.setVersion(MoPub.SDK_VERSION);
         mediationMetaData.set("adapter_version", UnityAdsAdapterConfiguration.ADAPTER_VERSION);
         mediationMetaData.commit();
+
+        MetaData headerBiddingMeta = new MetaData(context);
+        headerBiddingMeta.setCategory("headerbidding");
+        headerBiddingMeta.set("mode", "enabled");
+        headerBiddingMeta.commit();
```
Comment (MileyHollenberg) [inline UnityAds/src/main/java/com/mopub/mobileads/UnityRouter.java:81]: Are we expecting to have HB enabled by default? Might want to remove this if it was just for testing purposes

**Comment by MileyHollenberg** [comment]
Overall a good idea to consolidate the adapters into one. Please make sure not to touch any unrelated files like the libs of other providers which are being removed in this PR and also make sure to exclude any testing specific code.

**Pull request closed without merging**